lib/ocpp/v201/smart_charging.cpp

```diff
@@ -77,36 +77,58 @@ ProfileValidationResultEnum SmartChargingHandler::validate_tx_profile(const Char
  * - K01.FR.20
  * - K01.FR.34
  * - K01.FR.43
- * - K01.FR.45
  * - K01.FR.48
  */
-ProfileValidationResultEnum SmartChargingHandler::validate_profile_schedules(const ChargingProfile& profile) const {
-    auto schedules = profile.chargingSchedule;
+ProfileValidationResultEnum
+SmartChargingHandler::validate_profile_schedules(ChargingProfile& profile,
+                                                 std::optional<EvseInterface*> evse_opt) const {
+    auto schedules = &profile.chargingSchedule;
 
-    for (auto schedule : schedules) {
+    for (ChargingSchedule& schedule : *schedules) {
         // A schedule must have at least one chargingSchedulePeriod
         if (schedule.chargingSchedulePeriod.empty()) {
             return ProfileValidationResultEnum::ChargingProfileNoChargingSchedulePeriods;
         }
 
-        auto charging_schedule_period = schedule.chargingSchedulePeriod[0];
-
         for (auto i = 0; i < schedule.chargingSchedulePeriod.size(); i++) {
+            auto charging_schedule_period = &schedule.chargingSchedulePeriod[i];
```

Comment:
```suggestion
            const auto& charging_schedule_period = schedule.chargingSchedulePeriod[i];
```

Reply:
This can't be const since we modify its `numberPhases` for FR 49.

Reply:
Yep you are right, I did not see that yet when making this comment. The suggestion to change it from pointer type to reference still stands though.